jupyter_server/gateway/managers.py

```diff
@@ -374,14 +374,14 @@ async def start_kernel(self, kernel_id=None, path=None, **kwargs):
             The API path (unicode, '/' delimited) for the cwd.
             Will be transformed to an OS path relative to root_dir.
         """
-        self.log.info('Request start kernel: kernel_id=%s, path="%s"', kernel_id, path)
+        self.log.info(f"Request start kernel: kernel_id={kernel_id}, path='{path}'")
 
         if kernel_id is None:
             if path is not None:
                 kwargs['cwd'] = self.cwd_for_path(path)
             kernel_name = kwargs.get('kernel_name', 'python3')
             kernel_url = self._get_kernel_endpoint_url()
-            self.log.debug("Request new kernel at: %s" % kernel_url)
+            self.log.debug(f"Request new kernel at: {kernel_url}")
```

Comment:
One reason some people prefer to use old-style `%` formatting for logging is that it defers the string interpolation operation until the method is called. If your logging setting is below `DEBUG` you then don't have to pay the cost of the string interpolation. With f-strings you have to pay the cost whether or not the method is called.

Not sure if that's a concern here, but something to note.
https://stackoverflow.com/q/54367975

Reply:
That's really good to know Dave - thank you.  I think this area is okay but this will certainly come in handy for more intense areas.  We could also use `isEnabledFor()` if the string construction is a worry.

Reply:
I think calling a potentially expensive function may be more costly than the string interpolation! YMMV.

Reply:
@dhirschfeld that is true, but doesn't apply in this case since it uses the `%` operator directly, instead of passing the arguments into the logging call. Good to share as a PSA though 👍

Reply:
true - good point @vidartf

Reply:
Thanks @vidartf - I got the impression that there was some subtle python behavior with the `%` operator that would defer the string's resolution.  Glad to see that there's nothing funky going on and, yes, the use of arguments, particularly when expensive, is a good way to go for debug logging.  But at least that's true across the languages.